### DISCUSSION TITLE
do a shift in arrayjoin?

### DISCUSSION THREAD

**Discussion by euzada** [libvips/libvips#2510] [Q&A]
Hi, I was successfully able to merge a large microscope images together using Python and pyvips. My pictures are names like x_y.jpg. I have almost 10% overlap between each image on x and y. I used hspacing and vspacing to remove the overlap, which works well.  

I notice a slight shift in between 2images in x and y (around 15pixel). I am not able to remove the shift using arrayjoin. Is there any other way to do it?

This is the code I used.

Thank you.

`import pyvips`

`tiles = [pyvips.Image.new_from_file(f"{x}_{y}.jpg", access="sequential")`
`         for y in range(tiles_down) for x in range(tiles_across)]`
`im = pyvips.Image.arrayjoin(tiles, across=tiles_across)`

`im.write_to_file("x.jpg")`

**Comment by jcupitt** [comment]
Hi @euzada,

Do you mean +15 pixels in x/y on each join? Sorry, arrayjoin can only do rectangular grids, not rotated grids. 

You need to use `merge` to join all your frames. I would join horizontally first to make a set of strips, then join the strips vertically in a second pass. It'll feather your joins as well, which can look nice.

If you have a very large number of frames, you'll need to join in blocks of eg. 10 x 10 images.
